src/rhsmlib/dbus/objects/products.py

```diff
@@ -0,0 +1,63 @@
+from __future__ import print_function, division, absolute_import
+
+# Copyright (c) 2017 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public License,
+# version 2 (GPLv2). There is NO WARRANTY for this software, express or
+# implied, including the implied warranties of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+# along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+#
+# Red Hat trademarks are not licensed under GPLv2. No permission is
+# granted to use or replicate Red Hat trademarks that are incorporated
+# in this software or its documentation.
+import dbus
+import json
+import logging
+
+from rhsmlib.dbus import constants, base_object, util, dbus_utils
+from rhsmlib.services.products import InstalledProducts
+
+from subscription_manager.injectioninit import init_dep_injection
+
+init_dep_injection()
+
+log = logging.getLogger(__name__)
+
+
+class ProductsDBusObject(base_object.BaseObject):
+    """
+    A D-Bus object interacting with subscription-manager to list
+    all installed products.
+    """
+    default_dbus_path = constants.PRODUCTS_DBUS_PATH
+    interface_name = constants.PRODUCTS_INTERFACE
+
+    def __init__(self, conn=None, object_path=None, bus_name=None):
+        super(ProductsDBusObject, self).__init__(conn=conn, object_path=object_path, bus_name=bus_name)
+
+    @util.dbus_service_method(
+        constants.PRODUCTS_INTERFACE,
+        in_signature='sa{sv}',
+        out_signature='s')
+    @util.dbus_handle_exceptions
+    def ListInstalledProducts(self, filter_string, proxy_options, sender=None):
+
+        # FIXME: it is probably not necessary to reinitialize whole deep injection here.
+        # We do it here, because cert CertSorter has to be reinitialized.
```

Comment:
Yes, you are right. I should be more specific here. Imagine following scenario:

1. Start `rhsm-service`
2. Try to list installed product. You will get some list of installed product.
3. Attach to pool, enable some repository and install some package using yum. New product certificate will be installed (e.g. 70.pem)
4. Try to list installed product again. If deep injection wasn't reinitialized, then you would get same list of installed product (same as in step 2.), because deep injection tries to do memory caching and new product certificate (70.pem) would be ignored.